include/perf_ringbuffer.hpp

```diff
@@ -7,26 +7,28 @@
 
 #include "perf.hpp"
 
-typedef struct RingBuffer {
-  const char *start;
-  unsigned long offset;
-  size_t mask;
-  size_t meta_size; // size of the metadata page
-  size_t data_size; // size of data pages
-  size_t size;      // size of the entire region (data + metadata)
-  struct perf_event_mmap_page *region;
-  bool is_mirrored;
-  unsigned char *wrbuf;
-} RingBuffer;
-
-bool rb_init(RingBuffer *rb, struct perf_event_mmap_page *page, size_t size,
-             bool is_mirrored);
+#include <atomic>
```

Comment:
minor: I don't see usage of std::atomic 
Needed ?

Reply:
Bad split of changes on my change, I will remove this